File: src/Wollax.Cupel/ScoredItem.cs
namespace Wollax.Cupel;

/// <summary>
/// Pairs a context item with its computed relevance score.
/// </summary>
/// <param name="Item">The context item.</param>
/// <param name="Score">The computed relevance score, conventionally 0.0–1.0.</param>
public readonly record struct ScoredItem(ContextItem Item, double Score);

File: crates/cupel/src/model/scored_item.rs
use crate::model::ContextItem;

/// A value type associating a [`ContextItem`] with its computed relevance score.
#[derive(Debug, Clone)]
pub struct ScoredItem {
    /// The context item.
    pub item: ContextItem,
    /// The computed relevance score (conventionally in [0.0, 1.0]).
    pub score: f64,
}
